openexr-sys/c_wrapper/rust_ostream.cpp

```diff
@@ -0,0 +1,25 @@
+#include "rust_ostream.hpp"
+
+#include <stdexcept>
+
+using namespace IMATH_NAMESPACE;
+
+void RustOStream::write(const char c[], int n) {
+    int res = write_ptr(writer, c, n);
+    if (res != 0) {
+        throw std::runtime_error("error writing data");
```

Comment:
Round-tripping back into `io::Error` can be accomplished with a distinct `catch` clause in the CEXR code for `std::system_error`, and corresponding expansions of the interface. However, given that `std::system_error` alone should give us error messages, I don't think that needs to be addressed in this PR.